src/Either.ts

```diff
@@ -169,6 +181,17 @@ export class Right<L, A> {
   alt(fy: Either<L, A>): Either<L, A> {
     return this
   }
+  /**
+   * Lazy version of {@link alt}
+   * @since 1.6.0
+   * @param {(l: L) => Either<L, A>} fy - thunk
+   * @example
+   * assert.deepEqual(right(1).orElse(() => right(2)), right(1))
+   * @returns {Either<L, A>}
+   */
+  orElse(fy: (l: L) => Either<L, A>): Either<L, A> {
```

Comment:
As previously suggested, we could make this signature more general

```ts
orElse<M>(fy: (l: L) => Either<M, A>): Either<M, A>
```

Reply:
Sorry, missed that. Will fix

Reply:
@gcanti What about the `Right` side? It should return `this` where type of left side is `L`, not `M`.

Reply:
@raveclassic If the value is a `Right<L, A>` then is also a `Right<M, A>` for all `M`. You can return `this as any` (much like `map` in the `Left` case)